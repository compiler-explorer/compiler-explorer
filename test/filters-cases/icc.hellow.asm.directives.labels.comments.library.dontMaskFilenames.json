{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "main:"
    },
    {
      "labels": [],
      "source": null,
      "text": "..B1.1:                         # Preds ..B1.0"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.2:                                          #"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        pushq     %rbp                                          #2.12"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.4:                                          #"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        movq      %rsp, %rbp                                    #2.12"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.5:                                          #"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        andq      $-128, %rsp                                   #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        subq      $128, %rsp                                    #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        movl      $3, %edi                                      #2.12"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.8:                                          #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        call      __intel_new_proc_init                         #2.12"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.9:                                          #"
    },
    {
      "labels": [],
      "source": null,
      "text": "..B1.6:                         # Preds ..B1.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        stmxcsr   (%rsp)                                        #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 3,
        "mainsource": true
      },
      "text": "        movl      $.L_2__STRING.0, %edi                         #3.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 3,
        "mainsource": true
      },
      "text": "        xorl      %eax, %eax                                    #3.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        orl       $32832, (%rsp)                                #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        ldmxcsr   (%rsp)                                        #2.12"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.10:                                         #3.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 3,
        "mainsource": true
      },
      "text": "        call      printf                                        #3.1"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.11:                                         #"
    },
    {
      "labels": [],
      "source": null,
      "text": "..B1.2:                         # Preds ..B1.6"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 4,
        "mainsource": true
      },
      "text": "        movl      $.L_2__STRING.1, %edi                         #4.3"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 4,
        "mainsource": true
      },
      "text": "        xorl      %eax, %eax                                    #4.3"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.12:                                         #4.3"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 4,
        "mainsource": true
      },
      "text": "        call      printf                                        #4.3"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.13:                                         #"
    },
    {
      "labels": [],
      "source": null,
      "text": "..B1.3:                         # Preds ..B1.2"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 5,
        "mainsource": true
      },
      "text": "        xorl      %eax, %eax                                    #5.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 5,
        "mainsource": true
      },
      "text": "        movq      %rbp, %rsp                                    #5.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 5,
        "mainsource": true
      },
      "text": "        popq      %rbp                                          #5.1"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.15:                                         #"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 5,
        "mainsource": true
      },
      "text": "        ret                                                     #5.1"
    },
    {
      "labels": [],
      "source": null,
      "text": "..___tag_value_main.19:                                         #"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L_2__STRING.0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   72"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   101"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   108"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   108"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   111"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   32"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   119"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   111"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   114"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   108"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   100"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   0"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L_2__STRING.1:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   109"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   111"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   111"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   10"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   0"
    }
  ],
  "labelDefinitions": {
    "..B1.1": 2,
    "..B1.2": 23,
    "..B1.3": 29,
    "..B1.6": 14,
    "..___tag_value_main.10": 20,
    "..___tag_value_main.11": 22,
    "..___tag_value_main.12": 26,
    "..___tag_value_main.13": 28,
    "..___tag_value_main.15": 33,
    "..___tag_value_main.19": 35,
    "..___tag_value_main.2": 3,
    "..___tag_value_main.4": 5,
    "..___tag_value_main.5": 7,
    "..___tag_value_main.8": 11,
    "..___tag_value_main.9": 13,
    ".L_2__STRING.0": 36,
    ".L_2__STRING.1": 49,
    "main": 1
  }
}
