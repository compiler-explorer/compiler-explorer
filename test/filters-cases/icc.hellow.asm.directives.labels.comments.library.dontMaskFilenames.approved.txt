{
  "asm": [
    {
      "labels": [],
      "source": null,
      "text": "main:"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        pushq     %rbp                                          #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        movq      %rsp, %rbp                                    #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        andq      $-128, %rsp                                   #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        subq      $128, %rsp                                    #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        movl      $3, %edi                                      #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        call      __intel_new_proc_init                         #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        stmxcsr   (%rsp)                                        #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 3,
        "mainsource": true
      },
      "text": "        movl      $.L_2__STRING.0, %edi                         #3.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 3,
        "mainsource": true
      },
      "text": "        xorl      %eax, %eax                                    #3.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        orl       $32832, (%rsp)                                #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 2,
        "mainsource": true
      },
      "text": "        ldmxcsr   (%rsp)                                        #2.12"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 3,
        "mainsource": true
      },
      "text": "        call      printf                                        #3.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 4,
        "mainsource": true
      },
      "text": "        movl      $.L_2__STRING.1, %edi                         #4.3"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 4,
        "mainsource": true
      },
      "text": "        xorl      %eax, %eax                                    #4.3"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 4,
        "mainsource": true
      },
      "text": "        call      printf                                        #4.3"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 5,
        "mainsource": true
      },
      "text": "        xorl      %eax, %eax                                    #5.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 5,
        "mainsource": true
      },
      "text": "        movq      %rbp, %rsp                                    #5.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 5,
        "mainsource": true
      },
      "text": "        popq      %rbp                                          #5.1"
    },
    {
      "labels": [],
      "source": {
        "file": "-",
        "line": 5,
        "mainsource": true
      },
      "text": "        ret                                                     #5.1"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L_2__STRING.0:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   72"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   101"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   108"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   108"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   111"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   32"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   119"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   111"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   114"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   108"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   100"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   0"
    },
    {
      "labels": [],
      "source": null,
      "text": ".L_2__STRING.1:"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   109"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   111"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   111"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   10"
    },
    {
      "labels": [],
      "source": null,
      "text": "        .byte   0"
    }
  ],
  "labelDefinitions": {
    ".L_2__STRING.0": 21,
    ".L_2__STRING.1": 34,
    "main": 1
  }
}
